### ISSUE TITLE
Performance benchmarks section of the README is super out of date

### ISSUE THREAD

**Issue by glwagner** [CliMA/Oceananigans.jl#3684]
@simone-silvestri can I convince you to rewrite this section with updated benchmarks, and include results for distributed systems?

https://github.com/CliMA/Oceananigans.jl?tab=readme-ov-file#performance-benchmarks

**Comment by simone-silvestri** [comment]
Sounds good. Indeed those results are quite old

**Comment by ali-ramadhan** [comment]
Yeah the plot is from v0.5.x or something lol.

Not sure if CPU -> GPU speedups are as relevant anymore? Well, maybe a lot of people are still on CPUs. The speedup can be much larger with WENO, but it's all very CPU and GPU dependent. I've seen ~8x and ~2000x speedups for the same benchmark.

Also not sure if there are any good metrics to quantify performance. But @simone-silvestri's  scaling plots would probably be great to include.

**Comment by glwagner** [comment]
Right! The CPU -> GPU speed up has limited value, because the CPU is not a good reference / is not optimized. More useful would be a comparison with other models, somehow, but this is always challenging, especially for complicated models. Another useful piece of information would compare performance for difference GPUs.

This blog post (which advertises the advantages of liquid cooled GPUs) is relevant and includes some Oceananigans benchmarks:

https://www.markiiisys.com/blog/liquid-cooled-workstation-advantageous-for-big-ocean-simulations-exceptional-for-small-tasks-oregon-state-mark-iii-nvidia-supermicro/
